src/tools/swix/swcore/CompilerBackend/Nuget/NugetBackendCompiler.cs

```diff
@@ -53,7 +53,10 @@ public override void Generate(IEnumerable<Intermediate> intermediates, string ou
                 return;
             }
 
-            FileTransfer packageTransfer = FileTransfer.Create(null, Path.GetTempFileName(), outputPath, "NugetPackage", true);
+            int lastIndex = outputPath.LastIndexOf('.');
```

Comment:
I don't think the compiler should be messing with the output name internally. It would prevent you from being able to create whatever name you wanted. If this is done anywhere, it could be done in the msbuild targets.